src/components/TodosFilter.tsx

```diff
@@ -0,0 +1,80 @@
+import { useContext } from 'react';
+import { TodosContext } from './TodosContext';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+export const TodosFilter: React.FC = () => {
+  const {
+    todos,
+    handleStatus,
+    selectedStatus,
+    setTodos,
+  } = useContext(TodosContext);
+
+  const ItemsLeft = () => {
```

Comment:
The `ItemsLeft` function is being created each time `TodosFilter` re-renders. Consider using `useMemo` hook to prevent unnecessary calculations on each render.